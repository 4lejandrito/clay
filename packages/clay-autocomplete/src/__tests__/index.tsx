/**
 * Â© 2019 Liferay, Inc. <https://liferay.com>
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

import ClayAutocomplete from '..';
import React from 'react';
import {cleanup, render} from 'react-testing-library';

describe('ClayAutocomplete', () => {
	afterEach(cleanup);

	it('renders', () => {
		render(<ClayAutocomplete />);

		expect(document.body).toMatchSnapshot();
	});

	it('renders Autocomplete with other markup component', () => {
		const MyMarkup = React.forwardRef<
			HTMLDivElement,
			React.HTMLAttributes<HTMLDivElement>
		>(({children, ...otherProps}, ref) => (
			<div
				{...otherProps}
				className="form-control form-control-tag-group"
				ref={ref}
			>
				{children}
			</div>
		));

		render(<ClayAutocomplete component={MyMarkup} />);

		expect(document.body).toMatchSnapshot();
	});
});
