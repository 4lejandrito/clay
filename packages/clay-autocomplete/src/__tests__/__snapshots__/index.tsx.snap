// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayAutocomplete renders 1`] = `
<body>
  <div>
    <div
      class="input-group"
    >
      <div
        class="input-group-item"
      />
    </div>
  </div>
</body>
`;

exports[`ClayAutocomplete renders Autocomplete with other markup component 1`] = `
<body>
  <div>
    <div
      class="form-control form-control-tag-group"
    />
  </div>
</body>
`;
